ZFitter/bin/ZFitter.cpp

```diff
@@ -1166,9 +1207,1277 @@ int main(int argc, char **argv) {
+  
+  
+  
+  // Get trees
+  std::cout << std::endl;
+ 
+  /* 
+  TChain* ntu_DA = new TChain("simpleNtupleEoverP/SimpleNtupleEoverP");
+  FillChain(ntu_DA,"inputDATA.txt");
+  std::cout << "     DATA: " << std::setw(8) << ntu_DA->GetEntries() << " entries" << std::endl;
+  
+  TChain* ntu_MC = new TChain("simpleNtupleEoverP/SimpleNtupleEoverP");
+  FillChain(ntu_MC,"inputMC.txt");
+  std::cout << "REFERENCE: " << std::setw(8) << ntu_MC->GetEntries() << " entries" << std::endl;
+  */  
+
+  if (data->GetEntries() == 0 || mc->GetEntries() == 0 )
+  {
+    std::cout << "Error: At least one file is empty" << std::endl; 
+    return -1;
+  }
+  
+  
+  
+  // Set branch addresses
+  int runNumber;
+  int runTime;
+  int nPU;
+  float avgLCSCEle[3], seedLaserAlphaSCEle1, etaSCEle[3], phiSCEle[3], energySCEle[3], esEnergySCEle[3], pAtVtxGsfEle[3], energySCEle_corr[3];
+  int seedXSCEle[3], seedYSCEle[3];//, seedZside;
+  
+  data->SetBranchStatus("*",0);
+  data->SetBranchStatus("runNumber",1);  
+  data->SetBranchStatus("runTime",1);
+  data->SetBranchStatus("nPU",1);
+  data->SetBranchStatus("avgLCSCEle",1);
+  data->SetBranchStatus("seedLaserAlphaSCEle1",1);
+  //  data->SetBranchStatus("ele1_EOverP",1);
+  data->SetBranchStatus("etaSCEle",1);
+  data->SetBranchStatus("phiSCEle",1);
+  data->SetBranchStatus("energySCEle",1);
+  data->SetBranchStatus("energySCEle_corr",1);
+  data->SetBranchStatus("esEnergySCEle",1);
+  data->SetBranchStatus("pAtVtxGsfEle",1);
+  data->SetBranchStatus("seedXSCEle",1);
+  data->SetBranchStatus("seedYSCEle",1);
+  //  data->SetBranchStatus("ele1_seedZside",1);
+    
+  data->SetBranchAddress("runNumber", &runNumber);  
+  data->SetBranchAddress("runTime", &runTime);
+  data->SetBranchAddress("nPU", &nPU);
+  data->SetBranchAddress("avgLCSCEle", &avgLCSCEle[0]);
+  data->SetBranchAddress("seedLaserAlphaSCEle1", &seedLaserAlphaSCEle1);
+  //  data->SetBranchAddress("ele1_EOverP", &EoP);
+  data->SetBranchAddress("etaSCEle", &etaSCEle);
+  data->SetBranchAddress("phiSCEle", &phiSCEle);
+  if( useRegression < 1 )
+    data->SetBranchAddress("energySCEle", &energySCEle);
+  else
+    data->SetBranchAddress("energySCEle_corr", &energySCEle_corr);
+  data->SetBranchAddress("energySCEle", &energySCEle);
+  data->SetBranchAddress("esEnergySCEle", &esEnergySCEle);
+  data->SetBranchAddress("pAtVtxGsfEle", &pAtVtxGsfEle);
+  data->SetBranchAddress("seedXSCEle", &seedXSCEle);
+  data->SetBranchAddress("seedYSCEle", &seedYSCEle);
+  //  data->SetBranchAddress("ele1_seedZside", &seedZside);
+
+
+  mc->SetBranchStatus("*",0);
+  mc->SetBranchStatus("runNumber",1);  
+  mc->SetBranchStatus("runTime",1);
+  mc->SetBranchStatus("nPU",1);
+  mc->SetBranchStatus("avgLCSCEle",1);
+  mc->SetBranchStatus("seedLaserAlphaSCEle1",1);
+  //  mc->SetBranchStatus("ele1_EOverP",1);
+  mc->SetBranchStatus("etaSCEle",1);
+  mc->SetBranchStatus("phiSCEle",1);
+  mc->SetBranchStatus("energySCEle",1);
```

Comment:
you cannot put all these stuff in ZFitter, the monitoring should be provided by a dedicated class. ZFitter should be only an interface